noir-projects/aztec-nr/aztec/src/state_vars/public_immutable.nr

```diff
@@ -14,9 +14,9 @@ pub struct PublicImmutable<T, Context> {
 }
 // docs:end:public_immutable_struct
 
-impl<T, Context, let N: u32> Storage<N> for PublicImmutable<T, Context>
+impl<T, Context, let M: u32, let N: u32> Storage<N> for PublicImmutable<T, Context>
```

Comment:
There should be a comment here explaining that we're storing `WithHash<T>`, what it means (that we also store the hash, hence we need as many fields as withhash requires), and why we're doing it (because `T` can now be larger and result in a single private read).

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8022/commits/63595931d4a2c7ef9906ad1aaa881382e9e19a4d

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8022/commits/f8e8de322c567c32720d87bcb422b4a5981d6f50